lib/mail/encodings.rb

```diff
@@ -178,9 +178,8 @@ def Encodings.address_encode(address, charset = 'utf-8')
 
     def Encodings.encode_non_usascii(address, charset)
       return address if address.ascii_only? or charset.nil?
-      us_ascii = %Q{\x00-\x7f}
-      # Encode any non usascii strings embedded inside of quotes
-      address = address.gsub(/(".*?[^#{us_ascii}].*?")/) { |s| Encodings.b_value_encode(unquote(s), charset) }
+      # Encode all strings embedded inside of quotes
+      address = address.gsub(/(".*?")/) { |s| Encodings.b_value_encode(unquote(s), charset) }
```

Comment:
Good case to eliminate backtracking with the lazy `"[^"]*"` match instead of the non-greedy `".*?"`.